### ISSUE TITLE
[Test Missing] - Implement a test for the documentation

### ISSUE THREAD

**Issue by DEKHTIARJonathan** [tensorlayer/tensorlayer#504]
As highlighted by the PR #503, we miss a unittest for the documentation.

I created a new branch and initiated a commit: 8c95be to address this issue.
I'm not used to Sphinx, so if anyone wanna help on this contributions are welcome.

The main idea is to insure that we cannot merge PR that break the documentation as I unfortunately did in #502.

I thought about using this snippet:

https://stackoverflow.com/questions/33896196/how-to-test-the-validity-of-your-sphinx-documentation

```python
class DocTest(unittest.TestCase):
    source_dir = u'docs/source'
    config_dir = u'docs/source'
    output_dir = u'docs/build'
    doctree_dir = u'docs/build/doctrees'
    all_files = 1

    def test_html_documentation(self):
        app = Sphinx(self.source_dir,
                     self.config_dir,
                     self.output_dir,
                     self.doctree_dir,
                     buildername='html',
                     warningiserror=True,
        )
        app.build(force_all=self.all_files)
        # TODO: additional checks here if needed

    def test_text_documentation(self):
        # The same, but with different buildername
        app = Sphinx(self.source_dir,
                     self.config_dir,
                     self.output_dir,
                     self.doctree_dir,
                     buildername='text',
                     warningiserror=True,
        )
        app.build(force_all=self.all_files)
        # TODO:  additional checks if needed

    def tearDown(self):
        # TODO: clean up the output directory
        pass
```

**Comment by lgarithm** [comment]
I originally think about running the following command in travis ci 
```
cd docs
pip install -r requirements.txt
make html
```
as a simple test. But didn't do it, because tests were small, and I didn't want to increase test time.
Now I think it's reasonable to add a test for docs, since user experience is becoming more and more important.

The above snippet looks good, and more formal, I'm OK with it, too.

**Comment by DEKHTIARJonathan** [comment]
I would prefer using a unittest if it is possible. The best thing would have been to use a CI but rtfd do not provide one yet.

This could change in a near future:  https://github.com/rtfd/readthedocs.org/issues/1340

**Comment by DEKHTIARJonathan** [comment]
Implemented in #511
